src/lib/bootloader/mbr_update.rb

```diff
@@ -183,8 +156,8 @@ def activatable_partitions(disk)
       disk.partitions.reject { |p| p.id.is?(:swap, :bios_boot) }
     end
 
-    def extended_partition(disk)
-      part = activatable_partitions(disk).find { |p| p.type == Y2Storage::PartitionType::EXTENDED }
+    def extended_partition(partition)
+      part = partition.siblings.find { |p| p.type.is?(:extended) }
```

Comment:
ok, taking note about it. I just think that siblings make code more readable. I think it would be worth to mention there surprises in siblings documentation.